Collectd 5.3.1 DNS plugin does not build on RHEL5

When trying to build collectd 5.3.1 on RHEL 5.9, I receive the following error:

libtool: compile:  gcc -DHAVE_CONFIG_H -I. -DPREFIX=\"/usr\" -DCONFIGFILE=\"/etc/collectd.conf\" -DLOCALSTATEDIR=\"/var\" -DPKGLOCALSTATEDIR=\"/var/lib/collectd\" -DPIDFILE=\"/var/run/collectd.pid\" -DPLUGINDIR=\"/usr/lib64/collectd\" -DPKGDATADIR=\"/usr/share/collectd\" -Wall -Werror -O2 -g -m64 -mtune=generic "-DLT_LAZY_OR_NOW=RTLD_LAZY|RTLD_GLOBAL" -MT dns.lo -MD -MP -MF .deps/dns.Tpo -c dns.c  -fPIC -DPIC -o .libs/dns.o
dns.c: In function 'dns_run_pcap_loop':
dns.c:239: error: 'PCAP_ERROR' undeclared (first use in this function)
dns.c:239: error: (Each undeclared identifier is reported only once
dns.c:239: error: for each function it appears in.)
cc1: warnings being treated as errors
dns.c:246: warning: implicit declaration of function 'pcap_statustostr'
dns.c:246: warning: format '%s' expects type 'char _', but argument 3 has type 'int'
dns.c:254: warning: format '%s' expects type 'char *', but argument 3 has type 'int'
dns.c:272: error: 'PCAP_ERROR_IFACE_NOT_UP' undeclared (first use in this function)
dns.c: In function 'dns_child_loop':
dns.c:313: error: 'PCAP_ERROR_IFACE_NOT_UP' undeclared (first use in this function)
dns.c:319: error: 'PCAP_ERROR_BREAK' undeclared (first use in this function)
dns.c:320: warning: format '%s' expects type 'char *', but argument 3 has type 'int'
make[3]: *_\* [dns.lo] Error 1
make[3]: Leaving directory `/usr/src/redhat/BUILD/collectd-5.3.1/src'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory`/usr/src/redhat/BUILD/collectd-5.3.1/src'
make[1]: **\* [all] Error 2
make[1]: Leaving directory `/usr/src/redhat/BUILD/collectd-5.3.1/src'
make: **\* [all-recursive] Error 1
error: Bad exit status from /var/tmp/rpm-tmp.10923 (%build)

On this same machine, I managed to build 5.3.0 just fine. Could it be caused by b8f9530ad6ba3a55f129d259c705d9b5d4d3f4e5 ?

--- a/configure.in
+++ b/configure.in
@@ -3013,6 +3013,25 @@ then
 	AC_CHECK_HEADERS(pcap-bpf.h,,
 			 [with_libpcap="no (pcap-bpf.h not found)"])
 fi
+if test "x$with_libpcap" = "xyes"
+then
+	AC_CACHE_CHECK([whether libpcap has PCAP_ERROR_IFACE_NOT_UP],
+		       [c_cv_libpcap_have_pcap_error_iface_not_up],
+		       AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+[[[
+#include <pcap.h>
+]]],
+[[[
+  int val = PCAP_ERROR_IFACE_NOT_UP;
+]]]
+		       )],
+		       [c_cv_libpcap_have_pcap_error_iface_not_up="yes"],
+		       [c_cv_libpcap_have_pcap_error_iface_not_up="no"]))
+fi
+if test "x$c_cv_libpcap_have_pcap_error_iface_not_up" != "xyes"
+then
+		with_libpcap="no (pcap.h misses PCAP_ERROR_IFACE_NOT_UP)"
+fi
 AM_CONDITIONAL(BUILD_WITH_LIBPCAP, test "x$with_libpcap" = "xyes")
 # }}}
 

